tests/test_utils.py

```diff
@@ -66,7 +66,9 @@ def test_no_signature_builtins_not_devices() -> None:
     import tests.fake_beamline_misbehaving_builtins as fake_beamline
 
     devices = make_all_devices(fake_beamline)
-    assert not devices
+    assert not devices[0]
+    for exception in devices[1]:
```

Comment:
I just had a look at this, the device fake_beamlines don't create exceptions so I will need to add some devices in the fake_beamline that raises a exception when it does not connect

Reply:
I think we need the following test cases:
- All devices pass
- All devices fail
- Mixture
If you have to make a few small fake beamline modules to cover those then go for it